pkg/command/service/common.go

```diff
@@ -146,16 +150,99 @@ func GetIngressController(p *pkg.PerfParams) map[string]string {
+func GenerateHTMLOutput(csvPath string, outputPathPrefix string) (htmlPath string, err error) {
+	htmlPath = outputPathPrefix + ".html"
+	err = utils.GenerateHTMLFile(csvPath, htmlPath)
+	if err != nil {
+		fmt.Printf("failed to generate HTML file and skip %s\n", err)
+		return "", err
+	}
+	return htmlPath, nil
+}
+
+// GenerateJSONOutput generates JSON output from the result
+func GenerateJSONOutput(result interface{}, outputPathPrefix string) (jsonPath string, err error) {
+	jsonData, err := json.Marshal(result)
+	if err != nil {
+		fmt.Printf("failed to generate json data and skip %s\n", err)
+		return "", err
+	}
+	jsonPath = outputPathPrefix + ".json"
+	err = utils.GenerateJSONFile(jsonData, jsonPath)
+	if err != nil {
+		fmt.Printf("failed to generate json file and skip %s\n", err)
+		return "", err
+	}
+	return jsonPath, nil
+}
+
+// GenerateOutputPathPrefix generates the prefix of output path, which can be combined with a suffix name(.csv) to form a complete path
+func GenerateOutputPathPrefix(inputsOutput string, outputFilenameFlag string) (pathPrefix string, err error) {
+	current := time.Now()
+	outputLocation, err := utils.CheckOutputLocation(inputsOutput)
+	if err != nil {
+		fmt.Printf("failed to check measure output location: %s\n", err)
+		return "", err
+	}
+	pathPrefix = filepath.Join(outputLocation, fmt.Sprintf("%s_%s", current.Format(DateFormatString), outputFilenameFlag))
+	return pathPrefix, nil
+}
+
+// GenerateOutput generates outputs according to flags(csvFlag, htmlFlag and josnFlag) from rows and result
+func GenerateOutput(inputsOutput string, outputFilenameFlag string, csvFlag bool, htmlFlag bool, jsonFlag bool, rows [][]string, result interface{}) error {
+	outputPathPrefix, err := GenerateOutputPathPrefix(inputsOutput, outputFilenameFlag)
+	if err != nil {
+		return err
+	}
+	if csvFlag && rows != nil {
+		// generate csv file from rows
+		csvPath, err := GenerateCSVOutput(rows, outputPathPrefix)
+		if err == nil {
+			fmt.Printf("Measurement saved in CSV file %s\n", csvPath)
+		}
+		// generate html file from csv file
+		if htmlFlag && csvPath != "" {
+			htmlPath, err := GenerateHTMLOutput(csvPath, outputPathPrefix)
+			if err == nil {
+				fmt.Printf("Visualized measurement saved in HTML file %s\n", htmlPath)
+			}
+		}
+	} else if htmlFlag {
+		fmt.Printf("HTML output needs CSV output, please reset CSV Flag.\n")
+	}
+	if jsonFlag {
+		// generate json file from result
+		jsonPath, err := GenerateJSONOutput(result, outputPathPrefix)
+		if err == nil {
+			fmt.Printf("Measurement saved in JSON file %s\n", jsonPath)
+		}
+	}
+	return nil
+}
+
 // deleteFile deletes a file from the filepath
 func deleteFile(filepath string) error {
-	_, fileError := os.Stat(filepath)
-	if fileError == nil {
-		removeError := os.Remove(filepath)
-		if removeError != nil {
-			fmt.Printf("remove %s error : %s\n", filepath, removeError)
-			return removeError
+	_, err := os.Stat(filepath)
+	if err == nil {
```

Comment:
I prefer the common Golang style:
```go
  err := doSomething()
    if err != nil {
    return err
  }
  // no errors so lets do the work
  doTheWork()
  return nil
}
```

Reply:
Thank you, your style is more standardized, I will fixed it.